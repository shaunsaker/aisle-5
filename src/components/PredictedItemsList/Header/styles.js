import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    borderBottomWidth: 1,
    borderBottomColor: styleConstants.colors.dividerColor,
  },
  firstColumn: {
    flex: 1,
    marginRight: styleConstants.dimensions.spacing.horizontal,
  },
  secondColumn: {
    flex: 2,
  },
  text: {
    ...styleConstants.fonts.types.paragraph,
    ...styleConstants.fonts.families.bold,
    textDecorationLine: 'underline',
  },
});

export default styles;
