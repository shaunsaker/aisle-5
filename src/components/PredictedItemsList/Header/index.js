import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styleConstants from '../../../styleConstants';
import styles from './styles';

import LinkText from '../../LinkText';

const propTypes = {
  headerItems: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      flex: PropTypes.number,
      fieldName: PropTypes.string,
    }),
  ).isRequired,
  handleHeaderItemPress: PropTypes.func.isRequired,
  height: PropTypes.number,
  headerTestIDPrefix: PropTypes.string,
};

const defaultProps = {};

const Header = ({ headerItems, handleHeaderItemPress, height, headerTestIDPrefix }) => {
  return (
    <View style={[styles.container, { height }]}>
      {headerItems.map((item, index) => {
        const isNotLastItem = index !== headerItems.length - 1;

        return (
          <View
            key={item.text}
            style={[
              styles.row,
              { flex: item.flex },
              { marginRight: isNotLastItem ? styleConstants.dimensions.spacing.horizontal : 0 },
            ]}
          >
            <LinkText
              text={item.text}
              handlePress={() => handleHeaderItemPress(item.fieldName)}
              testID={`${headerTestIDPrefix}${item.text}`}
            />

            <Icon name="swap-vert" style={styles.icon} />
          </View>
        );
      })}
    </View>
  );
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
