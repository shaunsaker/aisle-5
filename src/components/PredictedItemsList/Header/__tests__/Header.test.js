import React from 'react';
import renderer from 'react-test-renderer';

import Header from '..';
import HEADER_ITEMS from '../../headerItems';

describe('Header', () => {
  describe('renders', () => {
    const handleHeaderItemPress = jest.fn();
    const height = 51;

    it('renders with minimum required props', () => {
      const component = renderer.create(
        <Header
          headerItems={HEADER_ITEMS}
          handleHeaderItemPress={handleHeaderItemPress}
          height={height}
        />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
