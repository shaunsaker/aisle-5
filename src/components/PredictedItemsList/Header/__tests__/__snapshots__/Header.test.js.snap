// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#EDE8CA",
        "borderBottomColor": "#FFFFFF",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      },
      Object {
        "height": 51,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "flex": 1,
        },
        Object {
          "marginRight": 15,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="undefinedNAME"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#08293A",
              "fontFamily": "Muli-Bold",
              "fontSize": 15,
              "textDecorationLine": "underline",
            },
            undefined,
          ]
        }
      >
        NAME
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#212121",
            "fontSize": 24,
            "marginLeft": 7.5,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "flex": 2,
        },
        Object {
          "marginRight": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="undefinedCONSUMPTION"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#08293A",
              "fontFamily": "Muli-Bold",
              "fontSize": 15,
              "textDecorationLine": "underline",
            },
            undefined,
          ]
        }
      >
        CONSUMPTION
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#212121",
            "fontSize": 24,
            "marginLeft": 7.5,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
